bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/AuthenticationApi.java

```diff
@@ -0,0 +1,107 @@
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.netatmo.internal.api.data.NetatmoConstants.FeatureArea;
+import org.openhab.binding.netatmo.internal.api.data.NetatmoConstants.Scope;
+import org.openhab.binding.netatmo.internal.api.dto.AccessTokenResponse;
+import org.openhab.binding.netatmo.internal.config.ApiHandlerConfiguration.Credentials;
+import org.openhab.binding.netatmo.internal.handler.ApiBridgeHandler;
+import org.openhab.core.common.ThreadPoolManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AuthenticationApi} handles oAuth2 authentication and token refreshing
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+public class AuthenticationApi extends RestManager {
+    private static final URI OAUTH_URI = getApiBaseBuilder().path(PATH_OAUTH).build();
+
+    private final ScheduledExecutorService scheduler = ThreadPoolManager.getScheduledPool(SERVICE_PID);
+    private final Logger logger = LoggerFactory.getLogger(AuthenticationApi.class);
+
+    private @Nullable ScheduledFuture<?> refreshTokenJob;
+    private Optional<AccessTokenResponse> tokenResponse = Optional.empty();
+    private String scope = "";
+
+    public AuthenticationApi(ApiBridgeHandler bridge) {
+        super(bridge, FeatureArea.NONE);
+    }
+
+    public void authenticate(Credentials credentials, Set<FeatureArea> features) throws NetatmoException {
+        Set<FeatureArea> requestedFeatures = !features.isEmpty() ? features : FeatureArea.AS_SET;
+        scope = FeatureArea.toScopeString(requestedFeatures);
+        requestToken(credentials.clientId, credentials.clientSecret,
+                Map.of(USERNAME, credentials.username, PASSWORD, credentials.password, SCOPE, scope));
+    }
+
+    private void requestToken(String id, String secret, Map<String, String> entries) throws NetatmoException {
+        Map<String, String> payload = new HashMap<>(entries);
+        payload.putAll(Map.of(GRANT_TYPE, entries.keySet().contains(PASSWORD) ? PASSWORD : REFRESH_TOKEN, CLIENT_ID, id,
+                CLIENT_SECRET, secret));
+        disconnect();
+        AccessTokenResponse response = post(OAUTH_URI, AccessTokenResponse.class, payload);
+        refreshTokenJob = scheduler.schedule(() -> {
+            try {
+                requestToken(id, secret, Map.of(REFRESH_TOKEN, response.getRefreshToken()));
+            } catch (NetatmoException e) {
+                logger.warn("Unable to refresh access token : {}", e.getMessage());
+            }
+        }, 5 /* Math.round(response.getExpiresIn() * 0.8) */, TimeUnit.SECONDS);
```

Comment:
If this is the final hardcoded value now, I would propose moving it to a constant and deleting the comment. Currently it seems like a left-over and brings doubt about intentions.